src/main/java/org/spongepowered/common/mixin/api/mcp/scoreboard/CriterionMixin_API.java

```diff
@@ -28,19 +28,40 @@
 import net.minecraft.scoreboard.IScoreCriteria;
 import net.minecraft.scoreboard.ScoreCriteria;
 import net.minecraft.scoreboard.ScoreCriteriaColored;
+import net.minecraft.util.text.TextFormatting;
 import org.spongepowered.api.scoreboard.critieria.Criterion;
+import org.spongepowered.api.text.format.TextColor;
 import org.spongepowered.asm.mixin.Implements;
 import org.spongepowered.asm.mixin.Interface;
 import org.spongepowered.asm.mixin.Intrinsic;
 import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.Surrogate;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.common.registry.type.text.TextColorRegistryModule;
 
 import javax.annotation.Nullable;
+import java.util.Optional;
 
 @Mixin(value = {ScoreCriteriaColored.class, ScoreCriteria.class})
 @Implements(@Interface(iface = Criterion.class, prefix = "criterion$"))
 public abstract class CriterionMixin_API implements IScoreCriteria { // Trick to allow avoid shadowing, since multiple targets are used
 
-    @Nullable private String spongeId;
+    @Nullable
+    private String spongeId;
+    @Nullable
+    private TextFormatting format;
+
+    @Inject(method = "<init>", at = @At(value = "RETURN"))
+    @SuppressWarnings("InvalidInjectorMethodSignature")
+    private void onConstructed(String name, CallbackInfo ci) {
```

Comment:
I really don't like this - this is a complicated "hack" to basically avoid doing two mixins instead of one.

I would much prefer that this logic is split into two separate mixins. This mixin, before your changes, can stay for the common logic, but using `@Surrogate` to support two constructors in two separate classes when the real intention is to use it for potential variences in the LVT for the same method is just... bad.

How Mixin works is black magic to some. Our Mixin classes should not be. If there needs to be an injection to a specific class, give it its own mixin.

(Also, this is not an API implementing method, so `api$`)

Reply:
Can I implement the same interface with 2 different mixins ?

Reply:
Yes